### PR TITLE
Fix the parse function returning an incomplete iterator

### PR DESCRIPTION
**Pull request by philn**
Without this patch the parse function would stop iterating over the File as soon
as a ParseError is raised.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -343,12 +343,18 @@ mod tests {
...
     }
 
-    #[test]
-    fn color() {
-        let f = File::open("test-logs/color.log").expect("Failed to open log file");
+    fn parse_file(f: File) -> (Entry, usize) {
         let mut parsed = parse(f);
 
         let entry = parsed.next().expect("First entry missing");
+        // FIXME: the +1 shouldn't be needed here.
```
Comment (gdesmott) [inline src/lib.rs]: Why not? You return the first element of the iter and count the rest right?

Comment (philn) [reply src/lib.rs]: Oh yeah, facepalm :)

**Comment by philn** [comment]
I used the opportunity to simplify the `parse_corrupted()` test as well.

**Comment by gdesmott** [comment]
Merged, thanks for the fix.

Do you want me to push a new release?

**Comment by philn** [comment]
That would be much appreciated because gst-log-diff is almost unusable without this patch :D No rush though!

**Comment by gdesmott** [comment]
Here we go: https://crates.io/crates/gst-log-parser

`cargo publish/crate` are so easy to use it's really no trouble to push a new release. :)

**Pull request merged**